x/concentrated-liquidity/bench_test.go

```diff
@@ -191,48 +178,94 @@ func BenchmarkSwapExactAmountIn(b *testing.B) {
 			for i := 0; i < numberOfPositions; i++ {
 				lowerTick := currentTick - 100
 				upperTick := currentTick + 100
-				// Normalize lowerTick to be a multiple of tickSpacing
 				lowerTick = lowerTick + (tickSpacing - lowerTick%tickSpacing)
-				// Normalize upperTick to be a multiple of tickSpacing
 				upperTick = upperTick - upperTick%tickSpacing
-
 				tokenDesired0 := sdk.NewCoin(denom0, sdk.NewInt(maxAmountDeposited).MulRaw(5))
 				tokenDesired1 := sdk.NewCoin(denom1, sdk.NewInt(maxAmountDeposited).MulRaw(5))
 				tokensDesired := sdk.NewCoins(tokenDesired0, tokenDesired1)
-
 				accountIndex := rand.Intn(len(s.TestAccs))
-
 				account := s.TestAccs[accountIndex]
-
 				simapp.FundAccount(s.App.BankKeeper, s.Ctx, account, tokensDesired)
-
 				s.createPosition(accountIndex, poolId, tokenDesired0, tokenDesired1, lowerTick, upperTick)
 			}
 		}
 
+		setupPositions()
+		if shouldCreateFullRangePositions {
+			setupFullRangePositions()
+		}
+		if shouldConcentrate {
+			setupConcentratedPositions()
+		}
+
 		swapAmountIn := sdk.MustNewDecFromStr(amountIn).TruncateInt()
 		largeSwapInCoin := sdk.NewCoin(denomIn, swapAmountIn)
 
+		testFunc(b, &s, pool, largeSwapInCoin, currentTick)
+
+	}
+}
+
+func BenchmarkSwapExactAmountIn(b *testing.B) {
+	runBenchmark(b, func(b *testing.B, s *BenchTestSuite, pool types.ConcentratedPoolExtension, largeSwapInCoin sdk.Coin, currentTick int64) {
+		clKeeper := s.App.ConcentratedLiquidityKeeper
+		noError := func(err error) {
+			require.NoError(b, err)
+		}
+
 		liquidityNet, err := clKeeper.GetTickLiquidityNetInDirection(s.Ctx, pool.GetId(), largeSwapInCoin.Denom, sdk.NewInt(currentTick), sdk.Int{})
 		noError(err)
+		simapp.FundAccount(s.App.BankKeeper, s.Ctx, s.TestAccs[0], sdk.NewCoins(largeSwapInCoin))
+
+		b.StartTimer()
+
+		// System under test
+		_, err = clKeeper.SwapExactAmountIn(s.Ctx, s.TestAccs[0], pool, largeSwapInCoin, DefaultCoin1.Denom, sdk.NewInt(1), pool.GetSpreadFactor(s.Ctx))
+		noError(err)
```

Comment:
Let's stop timer before `noError`

Reply:
wait why, I actually think we should undo this